### PR TITLE
fix: exclude self deleting messages from backup

### PR DESCRIPTION
**Pull request by MohamadJaara**
----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [ ] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [ ] contains a reference JIRA issue number like `SQPIT-764`
  - [ ] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [ ] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues

self deleting messages are included in backups 

### Causes (Optional)

case not handled

### Solutions

do not copy messages and related data when the message is self delete

### Dependencies (Optional)

_If there are some other pull requests related to this one (e.g. new releases of frameworks), specify them here._

Needs releases with:

- [ ] GitHub link to other pull request

#### Test Coverage (Optional)

- [ ] I have added automated test to this contribution

#### How to Test

_Briefly describe how this change was tested and if applicable the exact steps taken to verify that it works as expected._

### Notes (Optional)

_Specify here any other facts that you think are important for this issue._

### Attachments (Optional)

_Attachments like images, videos, etc. (drag and drop in the text box)_

----
#### PR Post Submission Checklist for internal contributors (Optional)

 - [ ] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

 - [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

**Review by tmspzz — APPROVED** [review]

**Review by vitorhugods — CHANGES_REQUESTED** [review]
Test for these?

`DatabaseImporterTest` has the tooling for that

File: persistence/src/commonTest/kotlin/com/wire/kalium/persistence/backup/DatabaseExporterTest.kt [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+            id = "OTHER_MENTIONING_SELF",
+            conversationId = TEST_CONVERSATION_1.id,
+            senderUserId = ORIGINAL_MESSAGE_SENDER.id,
+            content = MessageEntityContent.Text(
+                messageBody = "@${SELF_USER.name} @${OTHER_USER_2.name}", mentions = listOf(SELF_MENTION)
+            )
+        )
+    }
+
+
```
Comment (vitorhugods) [inline persistence/src/commonTest/kotlin/com/wire/kalium/persistence/backup/DatabaseExporterTest.kt:157]: Nit:

```suggestion
```

File: persistence/src/commonTest/kotlin/com/wire/kalium/persistence/backup/DatabaseExporterTest.kt [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+            backupDB.messageDAO.getMessageById(selfDeleteMessage.id, selfDeleteMessage.conversationId)?.also {
+                fail("Message should be null")
+            }
+            backupDB.messageDAO.getMessageById(normalMessage.id, normalMessage.conversationId)?.also {
+                assertEquals(normalMessage, it)
+            } ?: fail("Message should not be null")
+        }
+    }
+
+
```
Comment (vitorhugods) [inline persistence/src/commonTest/kotlin/com/wire/kalium/persistence/backup/DatabaseExporterTest.kt:87]: Nit:

```suggestion

```

File: persistence/src/commonTest/kotlin/com/wire/kalium/persistence/BaseDatabaseTest.kt [hunk 3]
```diff
@@ -32,13 +33,21 @@ expect open class BaseDatabaseTest() {
...
 
-    actual fun createDatabase(
+    fun platformDBData(userId: UserIDEntity): PlatformDatabaseData
+
+    fun createDatabase(
         userId: UserIDEntity,
         passphrase: UserDBSecret?,
         enableWAL: Boolean
     ): UserDatabaseBuilder
+
```
Comment (vitorhugods) [inline persistence/src/commonTest/kotlin/com/wire/kalium/persistence/BaseDatabaseTest.kt:51]:
```suggestion
```

**Pull request merged**